lib/btree2/test.c

```diff
@@ -0,0 +1,66 @@
+/****************************************************************************
+ *
+ * MODULE:       kdtree test
+ * AUTHOR(S):    林永 ynkan
+ * PURPOSE:      test the kdtree.c method for logical operations such as
+ *               kdtree_create=>kdtree_insert=>kdtree_dnn=>kdtree_remove=>kdtree_destroy
+ *               loop executionbalanced tree
+ *               See https://github.com/OSGeo/grass/issues/4779
+ * COPYRIGHT:    (C) 2024 by the GRASS Development Team
+ *
+ *               This program is free software under the GNU General Public
+ *               License (>=v2). Read the file COPYING that comes with GRASS
+ *               for details.
+ *
+ *****************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <time.h>
+#include "kdtree.h"
+
+#define MAX_POINTS 5
+
+void generate_random_points(double points[][2], int num_points)
+{
+    for (int i = 0; i < num_points; i++) {
+        points[i][0] = (rand() % 2000) / 100.0;
+        points[i][1] = (rand() % 2000) / 100.0;
+    }
+}
+
+int main(int argc, char *argv[])
```

Comment:
The CIs error at the `-Wunused-parameter` warning, this should solve that:

```suggestion
int main(void)
```

Reply:
Thanks! 
Now we'll see the segfault next...